Failing on coverage:

```
TOTAL                                                            17952      5    99%
Coverage failure: total of 99 is less than fail-under=100
Error: Process completed with exit code 2.
```

--- a/axelrod/game.py
+++ b/axelrod/game.py
@@ -23,7 +23,7 @@ class AsymmetricGame(object):
     """
 
     # pylint: disable=invalid-name
-    def __init__(self, A: np.array, B: np.array, **attributes) -> None:
+    def __init__(self, A: np.array, B: np.array, **characteristics) -> None:
         """
         Creates an asymmetric game from two matrices.
 
@@ -33,8 +33,8 @@ class AsymmetricGame(object):
             the payoff matrix for player A.
         B: np.array
             the payoff matrix for player B.
-        **attributes
-            optional attributes for the game. Used
+        **characteristics
+            optional characteristics detailing features of the game. Used
             to ensure strategies used with the game are valid.
         """
 
@@ -47,7 +47,7 @@ class AsymmetricGame(object):
         self.A = A
         self.B = B
 
-        self.attributes = attributes
+        self.characteristics = characteristics
 
         self.scores = {
             pair: self.score(pair) for pair in ((C, C), (D, D), (C, D), (D, C))
@@ -84,25 +84,25 @@ class AsymmetricGame(object):
         return (self.A[row][col], self.B[row][col])
 
     @property
-    def attributes(self):
-        return self._attributes
+    def characteristics(self):
+        return self._characteristics
 
-    @attributes.setter
-    def attributes(self, attributes):
+    @characteristics.setter
+    def characteristics(self, characteristics):
         """
-        Adds or changes game attributes.
+        Adds or changes game characteristics.
 
         Parameters
         ----------
-        attributes: dict
-            Attributes to add to the game. If the added
-            attribute already exists, it will overwrite the
+        characteristics: dict
+            characteristics to add to the game. If the added
+            characteristic already exists, it will overwrite the
             previous value.
         """
         try:
-            self._attributes = {**self._attributes, **attributes}
+            self._characteristics = {**self._characteristics, **characteristics}
         except AttributeError:
-            self._attributes = attributes
+            self._characteristics = characteristics
 
     def __repr__(self) -> str:
         return "Axelrod game with matrices: {}".format((self.A, self.B))
@@ -126,7 +126,7 @@ class Game(AsymmetricGame):
         The numerical score attribute to all combinations of action pairs.
     """
 
-    def __init__(self, r: Score = 3, s: Score = 0, t: Score = 5, p: Score = 1, **attributes) -> None:
+    def __init__(self, r: Score = 3, s: Score = 0, t: Score = 5, p: Score = 1, **characteristics) -> None:
         """Create a new game object.
 
         Parameters
@@ -146,12 +146,12 @@ class Game(AsymmetricGame):
         """
         A = np.array([[r, s], [t, p]])
 
-        default_attributes = {
-            'has_RPST': True
+        default_characteristics = {
+            'has_RPST': hasattr(self, 'RPST')
         }
-        attributes = {**default_attributes, **attributes}
+        characteristics = {**default_characteristics, **characteristics}
 
-        super().__init__(A, A.transpose(), **attributes)
+        super().__init__(A, A.transpose(), **characteristics)
 
     def RPST(self) -> Tuple[Score, Score, Score, Score]:
         """Returns game matrix values in Press and Dyson notation."""

--- a/axelrod/match.py
+++ b/axelrod/match.py
@@ -117,11 +117,11 @@ class Match(object):
         """Ensure that players are passed the match attributes"""
         newplayers = []
         # ensure the game satisfies the player assumptions
-        # note the game size attribute is added here because the player
+        # note the game size characteristic is added here because the player
         # and coplayer may have different game sizes if the game is asymmetric!
-        players[0].check_assumptions({**self.game.attributes, 'actions_size': self.game.A.shape[0]},
+        players[0].check_assumptions({**self.game.characteristics, 'actions_size': self.game.A.shape[0]},
                                       raise_error=self.strict_player_checking)
-        players[1].check_assumptions({**self.game.attributes, 'actions_size': self.game.B.shape[0]},
+        players[1].check_assumptions({**self.game.characteristics, 'actions_size': self.game.B.shape[0]},
                                       raise_error=self.strict_player_checking)
 
         for player in players:

--- a/axelrod/player.py
+++ b/axelrod/player.py
@@ -258,15 +258,15 @@ class Player(object, metaclass=PostInitCaller):
     def update_history(self, play, coplay):
         self.history.append(play, coplay)
 
-    def check_assumptions(self, attributes: dict, raise_error: bool=True):
+    def check_assumptions(self, characteristics: dict, raise_error: bool=True):
         """
-        Compares the player assumptions to a dictionary of attributes.
+        Compares the player assumptions to a dictionary of game characteristics.
         Generates a warning or error if an assumption is not fulfilled.
 
         Parameters:
         -----------
-        attributes: dict
-            The dictionary of attributes to compare the player's assumptions to.
+        characteristics: dict
+            The dictionary of game characteristics to compare the player's assumptions to.
         raise_error: bool, default True
             If True, raises an error if the assumption is violated. Else,
             just generate a warning.
@@ -274,36 +274,36 @@ class Player(object, metaclass=PostInitCaller):
 
         for key, value in self.classifier.get('assumptions', {}).items():
             msg = None
-            if key not in attributes.keys():
+            if key not in characteristics.keys():
                 msg = ("Player {} assumes that "
                        "the game has the attribute {}, "
                        "but the game does not declare this attribute."
                        "".format(self.name, key))
-            elif value != attributes[key]:
+            elif value != characteristics[key]:
                 msg = ("Player {} assumes that the game attribute "
                        "{} is set to {}, but it is actually set to {}."
-                       "".format(self.name, key, value, attributes[key]))
+                       "".format(self.name, key, value, characteristics[key]))
 
             if msg is not None:
                 if raise_error:
                     raise RuntimeError(msg)
                 warnings.warn(msg + " The strategy may not behave as expected.")
 
-    def assumptions_satisfy(self, attributes: dict) -> bool:
+    def assumptions_satisfy(self, characteristics: dict) -> bool:
         """
-        Compares the player assumptions to a dictionary of attributes.
+        Compares the player assumptions to a dictionary of game characteristics.
         Returns True if the player assumptions are all satisfied by
-        these attributes, and False otherwise.
+        these characteristics, and False otherwise.
 
         Parameters:
         -----------
-        attributes: dict
-            The dictionary of attributes to compare the player's assumptions to.
+        characteristics: dict
+            The dictionary of game characteristics to compare the player's assumptions to.
 
         Returns
         -------
         bool
-            A boolean of whether or not the attributes satisfy the player's
+            A boolean of whether or not the game characteristics satisfy the player's
             assumptions.
         """
 
@@ -311,7 +311,7 @@ class Player(object, metaclass=PostInitCaller):
         # around as check_assumptions needs finer grained understanding of
         # the assumptions to produce useful error messages
         try:
-            self.check_assumptions(attributes, raise_error=True)
+            self.check_assumptions(characteristics, raise_error=True)
         except RuntimeError:
             return False
         return True

--- a/axelrod/tests/strategies/test_player.py
+++ b/axelrod/tests/strategies/test_player.py
@@ -355,18 +355,32 @@ class TestPlayerClass(unittest.TestCase):
         player = axl.MockPlayer(classifier={'assumptions': {'foo': True, 'bar': 3}})
 
         # these should pass without errors/warnings
-        player.check_assumptions({'foo': True, 'bar': 3})
-        player.check_assumptions({'foo': True, 'bar': 3, 'baz': []})
+        player.check_assumptions({'foo': True, 'bar': 3})  # correct characteristics
+        player.check_assumptions({'foo': True, 'bar': 3, 'baz': []})  # extraneous characteristic
 
         with self.assertRaises(RuntimeError):
-            player.check_assumptions({'foo': True})
+            player.check_assumptions({'foo': True})  # missing characteristic
         with self.assertRaises(RuntimeError):
-            player.check_assumptions({'foo': True, 'bar': 5})
+            player.check_assumptions({'foo': True, 'bar': 5})  # invalid charateristic value
 
         with self.assertWarns(UserWarning):
-            player.check_assumptions({'foo': True}, raise_error=False)
+            player.check_assumptions({'foo': True}, raise_error=False)  # missing characteristic
         with self.assertWarns(UserWarning):
-            player.check_assumptions({'foo': True, 'bar': 5}, raise_error=False)
+            player.check_assumptions({'foo': True, 'bar': 5}, raise_error=False)  # invalid charateristic value
+
+    def test_assumptions_satisfy(self):
+        """
+        Tests that the assumptions_satisfy() method works as intended.
+        It is a wrapper around check_assumptions() so the actual assumption
+        testing logic is checked more thoroughly there.
+        """
+        player = axl.MockPlayer(classifier={'assumptions': {'foo': True, 'bar': 3}})
+
+        self.assertEqual(player.assumptions_satisfy({'foo': True, 'bar': 3}), True)  # correct characteristics
+        self.assertEqual(player.assumptions_satisfy({'foo': True, 'bar': 3, 'baz': []}), True)  # extraneous characteristic
+        self.assertEqual(player.assumptions_satisfy({'foo': True}), False)  # missing characteristic
+        self.assertEqual(player.assumptions_satisfy({'foo': True, 'bar': 5}), False)  # invalid charateristic value
+
 
 
 

